pydantic_ai_slim/pydantic_ai/mcp.py

```diff
@@ -517,3 +548,29 @@ async def main():
     @property
     def _transport_client(self):
         return streamablehttp_client  # pragma: no cover
+
+
+ToolResult = (
+    str | BinaryContent | dict[str, Any] | list[Any] | Sequence[str | BinaryContent | dict[str, Any] | list[Any]]
+)
+"""The result type of a tool call."""
+
+CallToolFn = Callable[[str, dict[str, Any], dict[str, Any] | None], Awaitable[ToolResult]]
```

Comment:
`Func` suffix for consistency with elsewhere:

```suggestion
CallToolFunc = Callable[[str, dict[str, Any], dict[str, Any] | None], Awaitable[ToolResult]]
```